edge-hub/core/test/Microsoft.Azure.Devices.Edge.Hub.Http.Test/RegistryControllerTest.cs

```diff
@@ -298,6 +298,38 @@ public async Task DeleteModuleOnBehalfOf_Success()
             Assert.Empty(actualResponseBytes);
         }
 
+        [Theory]
+        [InlineData("l4", "l3;l4", "l3", "l3;l4;l5", true, HttpStatusCode.OK)]
+        [InlineData("l4", "l2;l3;l4", "l2", "l2;l3;l4;l5", true, HttpStatusCode.OK)]
+        [InlineData("l4", "l4", "l4", "l4;l5", true, HttpStatusCode.OK)]
+        [InlineData("l3", "l4", "l4", "l4;l5", true, HttpStatusCode.Unauthorized)]
+        [InlineData("l4", "l3;l4", "l3", null, true, HttpStatusCode.Unauthorized)]
+        [InlineData("l4", "l3;l4", "l3", "l3;l4;l5", false, HttpStatusCode.Unauthorized)]
```

Comment:
Would be good to add the following cases for client-provided authchain:
repeated ID: "l4;l4;l4"
reversed ID: "l5;l4;l3"